File: Form Validator/Form Validator/ViewController.m
//
//  ViewController.m
//  Form Validator
//
//  Created by Keaton Swoboda on 3/9/15.
//  Copyright (c) 2015 The Iron Yard. All rights reserved.
//

#import "ViewController.h"
#import "FormValidator.h"

@interface ViewController ()<UITextFieldDelegate>

{
    FormValidator *validator;
}

@property (weak, nonatomic) IBOutlet UITextField *nameTextField;
@property (weak, nonatomic) IBOutlet UITextField *streetTextField;
@property (weak, nonatomic) IBOutlet UITextField *cityTextField;
@property (weak, nonatomic) IBOutlet UITextField *stateTextField;
@property (weak, nonatomic) IBOutlet UITextField *zipTextField;
@property (weak, nonatomic) IBOutlet UITextField *phoneTextField;



@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    validator = [[FormValidator alloc] init];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
 
}



-(BOOL)textFieldShouldReturn:(UITextField *)textField
{
    BOOL rc = NO;
    
    if (textField == self.nameTextField)
    {
       rc = [validator validateName:self.nameTextField.text];
        if (rc)
        {
            [textField resignFirstResponder];
            
        }
    }

  
    

    if (textField == self.streetTextField)
        {
          rc = [validator validateAddress:self.streetTextField.text];
            if (rc)
            {
                [textField resignFirstResponder];
                
            }
        }
    if (textField == self.cityTextField)
    {
        rc = [validator validateCity:self.cityTextField.text];
    }
    if (textField == self.stateTextField)
    {
        rc = [validator validateState:self.stateTextField.text];
    }
    if (textField == self.zipTextField)
    {
       rc = [validator validateZipCode:self.zipTextField.text];
    }
    if (textField == self.phoneTextField)
    {
       rc = [validator validatePhoneNumber:self.phoneTextField.text];
    }
    if (rc)
    {
        [textField resignFirstResponder];
    }
    

    return rc;
}




@end

File: SwiftBriefing/SwiftBriefing/ViewController.swift
//
//  ViewController.swift
//  SwiftBriefing
//
//  Created by Keaton Swoboda on 4/1/15.
//  Copyright (c) 2015 The Iron Yard. All rights reserved.
//

import UIKit

class ViewController: UIViewController, UITextFieldDelegate
{
    @IBOutlet var nameTextField : UITextField!
    @IBOutlet var passTextField : UITextField!
    @IBOutlet var authenticateButton : UIButton!
    @IBOutlet var briefingTextView : UITextView!
    @IBOutlet var goodMorningLabel : UILabel!
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        briefingTextView.text = ""
        nameTextField.delegate = self
        passTextField.delegate = self
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
    func textFieldShouldReturn(textField: UITextField) -> Bool
    {
        if textField == nameTextField
        {
            
        textField.resignFirstResponder()
        passTextField.becomeFirstResponder()
        }
        else
        {
            textField.resignFirstResponder()
            authenticateOnReturn()
            
        }
        return true
    }
    
    
    
    @IBAction func authButton(sender : AnyObject)
    {
       authenticateOnReturn()
        
        
    }
    func authenticateOnReturn()
    {
        if (nameTextField.text.isEmpty) || (passTextField.text.isEmpty)
        {
            goodMorningLabel.text = "Error, Error, Will self destruct"
            view.backgroundColor = UIColor.redColor()
            briefingTextView.backgroundColor = UIColor.redColor()
            briefingTextView.text = ""
        }
            
        else if !(nameTextField.text.isEmpty) || !(passTextField.text.isEmpty)
        {
            var agentNames = nameTextField.text.componentsSeparatedByString(" ")
            var lastNM = agentNames[1]
            goodMorningLabel.text = String(format:"Good Morning Angel %@", lastNM)
            briefingTextView.text = "Todays asignment is a risky one. You must locate a stolen Bengal Tiger and return it to the San Diego Zoo."
            passTextField.resignFirstResponder()
            view.backgroundColor = UIColor.greenColor()
            briefingTextView.backgroundColor = UIColor.greenColor()
        }
    }


}

